solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/indexing/SolrUpdateDriver.java

```diff
@@ -92,19 +86,25 @@ public void addDocument(IndexDocumentBean idb) throws IndexingException {
      */
     public void updateDocument(String operation, IndexDocumentBean idb) throws IndexingException {
         try {
-            SolrInputDocument sid = updateSolrClient.getBinder().toSolrInputDocument(idb);
-            for (String fieldName : sid.getFieldNames()) {
-                if (!ID_FIELD.equals(fieldName)) {
-                    SolrInputField inputField = sid.getField(fieldName);
-                    // Adding in each non-null field value, except the timestamp field which gets cleared
-                    // if not specified so that it always gets updated as part of a partial update
-                    if (inputField != null && (inputField.getValue() != null || UPDATE_TIMESTAMP.equals(fieldName))) {
-                        Map<String, Object> partialUpdate = new HashMap<>();
-                        partialUpdate.put(operation, inputField.getValue());
-                        sid.setField(fieldName, partialUpdate);
-                    }
+            SolrInputDocument sid = new SolrInputDocument();
+            Map<String, Object> fields = idb.getDynamicFields();
+            for (Entry<String, Object> field : fields.entrySet()) {
+                String fieldName = field.getKey();
+
+             // Avoid specifying the timestamp so it will get updated as part of this partial update
+                if (UPDATE_TIMESTAMP.equals(fieldName)) {
+                    continue;
+                }
+
+                Object value = field.getValue();
+
+                // Id field needs to be set like a non-partial update
+                if (ID.getSolrField().equals(fieldName)) {
+                    sid.addField(fieldName, value);
+                    continue;
                 }
             }
+
```

Comment:
I think a part is missing here
https://github.com/UNC-Libraries/Carolina-Digital-Repository/pull/634/files#diff-b26ca2d0f8ef083ca3f4b788413f0fb9R107